api/src/main/java/ee/tlu/evkk/api/controller/ContentExtractorController.java

```diff
@@ -32,8 +32,8 @@ public class CorsFilter extends OncePerRequestFilter {
     @Override
     protected void doFilterInternal(final HttpServletRequest request, final HttpServletResponse response,
                                     final FilterChain filterChain) throws ServletException, IOException {
-      response.addHeader("Access-Control-Allow-Origin", "*");
-      response.addHeader("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT, PATCH, HEAD");
+      response.addHeader("Access-Control-Allow-Origin", "http://localhost:9092");
```

Comment:
lokaalselt nii töötab aga serverisse buildides see aadress ei ole tema jaoks enam localhost:9092. Kuidagi peab saama panna kirja et samalt domeenilt aga teiselt pordilt (9092)

Reply:
äkki @mikktarvas teab kommenteerida, kuidas õige oleks? ma googeldades pole ka endiselt kindel

Reply:
Spring bootil on olemas sisse ehitatud CORS tugi. Peaks vältima ise filtri kirjutamist ja kasutama olemasolevaid tööriistasid.
https://spring.io/guides/gs/rest-service-cors/
CORS reeglid rakendatakse brauseri kontekstis ehk siis sellele aadressile mis brauserile välja paistab.
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

Reply:
@mikktarvas Ma proovisin seda CORSI Springi tööriistadega ümber thea, aga ma ei saanud seda toimima kuidagi. Viskab pidevalt "No 'Access-Control-Allow-Origin' header is present on the requested resource." errorit. Proovisin @CrossOrigini muuta ja proovisin ka eraldi Beanina teha.

Ma panen siia alla koodinäited ka. Võibolla leiad lahenduse.

@CrossOrigin(origins = "http://localhost:9092",
      methods = {GET, POST},
      allowedHeaders = {"Authorization", "Accept", "Origin", "X-Requested-With", "Content-Type", "Access-Control-Request-Headers", "Access-Control-Request-Method", "Allow-Origin"},
      exposedHeaders = {"Access-Control-Allow-Origin", "Access-Control-Allow-Credentials"},
      allowCredentials = "true", maxAge = 3600L)


   @Bean
    CorsConfigurationSource corsConfigurationSource() {
    CorsConfiguration configuration = new CorsConfiguration();
    configuration.setAllowedOrigins(Arrays.asList("http://localhost:9092"));
    configuration.setAllowedMethods(Arrays.asList("POST","GET"));
    configuration.setAllowCredentials(true);
    configuration.setAllowedHeaders(Arrays.asList("Accept", "Origin", "X-Requested-With", "Content-Type", "Access- 
    Control-Request-Headers", "Access-Control-Request-Method"));
    configuration.setExposedHeaders(Arrays.asList("Access-Control-Allow-Origin","Access-Control-Allow-Credentials"));
    configuration.setMaxAge(3600L);
    UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
    source.registerCorsConfiguration("/**", configuration);
    return source;

Lisasin mõlema puhul ka cors().and() turvakonfiguratsiooni.